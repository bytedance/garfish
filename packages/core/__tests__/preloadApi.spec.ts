import { mockStaticServer } from '@garfish/test-suite';
import assert from 'assert';
import Garfish from '../src/index';

const sleep = (time = 200) => {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve(false);
    }, time);
  });
};

describe('Core: preload plugin', () => {
  let GarfishInstance: Garfish;
  let container;
  const preloadSubAppEntry = './resources/preloadApp.html';

  mockStaticServer({
    baseDir: __dirname,
    timeConsuming: 200,
  });

  beforeEach(() => {
    container = document.createElement('div');
    container.setAttribute('id', 'container');
    document.body.appendChild(container);
    GarfishInstance = new Garfish({});
  });

  afterEach(() => {
    document.body.removeChild(container);
  });

  it('preload caches file success', async () => {
    const GarfishInstance = new Garfish({});
    let app: Awaited<ReturnType<typeof GarfishInstance.loadApp>> = null;
    GarfishInstance.run({
      disablePreloadApp: false,
      domGetter: '#container',
      apps: [
        {
          name: 'preload-app',
          entry: preloadSubAppEntry,
        },
      ],
    });
    GarfishInstance.preloadApp('preload-app');
    await sleep(410);
    GarfishInstance.loadApp('preload-app').then((res) => {
      app = res;
    });
    await sleep(50);
    assert(app, 'app should be loaded');
    expect(app).toBeDefined();
    await (app as any).mount();
    expect(
      document.body.querySelector('div[__garfishmockhtml__]'),
    ).toMatchSnapshot();
  });
});
