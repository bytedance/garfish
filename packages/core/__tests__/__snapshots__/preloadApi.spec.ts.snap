// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core: preload plugin preload caches file success 1`] = `
<div
  __garfishmockhtml__=""
>
  <div
    __garfishmockhead__=""
  >
    
    
    <title>
      vue sub app
    </title>
    
  
  </div>
  <div
    __garfishmockbody__=""
  >
    
    
    <div
      id="app"
    >
      <div>
        Hello World preload app
      </div>
    </div>
    
    
    <!--&lt;script src="./scripts/render.js"  execute by garfish&gt;&lt;/script&gt;-->
    
  


  </div>
</div>
`;
